gateway/handlers/external_auth.go

```diff
@@ -0,0 +1,37 @@
+package handlers
+
+import (
+	"context"
+	"net/http"
+	"time"
+)
+
+// MakeExternalAuthHandler make an authentication proxy handler
+func MakeExternalAuthHandler(next http.HandlerFunc, upstreamTimeout time.Duration, upstreamURL string, passBody bool) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		req, _ := http.NewRequest(http.MethodGet, upstreamURL, nil)
+
+		deadlineContext, cancel := context.WithDeadline(
```

Comment:
You could use `context.WithTimeout(context.Background(), upstreamTimeout)`

Reply:
That's a good point.